pypolychord/polychord.py

```diff
@@ -225,10 +226,430 @@ def wrap_prior(cube, theta):
+
+    <root> = <base_dir>/<file_root>
+
+    <root>.txt                                              (posteriors = True)
+        Weighted posteriors. Compatible with getdist. Each line contains:
+
+          weight, -2*loglikelihood, <physical parameters>, <derived parameters>
+
+        Note that here the weights are not normalised, but instead are chosen
+        so that the maximum weight is 1.0.
+
+    <root>_equal_weights.txt                                    (equals = True)
+        Weighted posteriors. Compatible with getdist. Each line contains:
+            1.0, -2*loglikelihood, <physical parameters>, <derived parameters>
+
+    <root>_dead.txt                                         (write_dead = True)
+        Dead points. Each line contains:
+            loglikelihood, <physical parameters>, <derived parameters>
+
+    <root>_phys_live.txt                                    (write_live = True)
+        Live points. Each line contains:
+            <physical parameters>, <derived parameters>, loglikelihood
+        Mostly useful for run-time monitoring.
+
+    <root>.resume
+        Resume file. Human readable.
+
+    <root>.stats
+        Final output evidence statistics
+
+    """
+
+    try:
+        from mpi4py import MPI
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+    except ImportError:
+        rank = 0
+
+    paramnames = kwargs.pop('paramnames', None)
+
+    default_kwargs = {
+        'nDerived': 0,
+        'prior': default_prior,
+        'dumper': default_dumper,
+        'nlive': nDims*25,
+        'num_repeats': nDims*5,
+        'nprior': -1,
+        'nfail': -1,
+        'do_clustering': True,
+        'feedback': 1,
+        'precision_criterion': 0.001,
+        'logzero': -1e30,
+        'max_ndead': -1,
+        'boost_posterior': 0.0,
+        'posteriors': True,
+        'equals': True,
+        'cluster_posteriors': True,
+        'write_resume': True,
+        'write_paramnames': False,
+        'read_resume': True,
+        'write_stats': True,
+        'write_live': True,
+        'write_dead': True,
+        'write_prior': True,
+        'maximise': False,
+        'compression_factor': np.exp(-1),
+        'synchronous': True,
+        'base_dir': 'chains',
+        'file_root': 'test',
+        'grade_dims': [nDims],
+        'nlives': {},
+        'seed': -1,
+    }
+    default_kwargs['grade_frac'] = [1.0]*len(default_kwargs['grade_dims'])
+
+    if not set(kwargs.keys()) <= set(default_kwargs.keys()):
+        raise TypeError(f"{__name__} got unknown keyword arguments {kwargs.keys() - default_kwargs.keys()}")
+    default_kwargs.update(kwargs)
+    kwargs = default_kwargs
+
+    try:
+        if rank == 0:
+            os.makedirs(kwargs['base_dir'])
+    except OSError:
+        pass
+
+    try:
+        if rank == 0:
+            os.makedirs(os.path.join(kwargs['base_dir'], 'clusters'))
+    except OSError:
+        pass
```

Comment:
I think no, as I believe the OSError expected here is that the directory already exists, and it may be the case that `base_dir` already exists, but `base_dir/clusters` does not.

I can do something neater with `pathlib.Path`, which can create several layers of directories at once, and can be told not to worry if the directory already exists

Reply:
Done, I've also converted the other uses of `os` to `pathlib`